dotCMS/src/main/java/com/dotcms/rest/IntegrityResource.java

```diff
@@ -373,81 +463,75 @@ public Response checkIntegrity(@Context HttpServletRequest httpServletRequest, @
             return response( "Error checking existence of integrity data" , true );
         }
 
-        try {
-
-            //Setting the process status
-            setStatus( httpServletRequest, endpointId, ProcessStatus.PROCESSING );
+        //Setting the process status
+        setStatus(httpServletRequest, endpointId, ProcessStatus.PROCESSING);
 
+        try {
             final PublishingEndPoint endpoint = APILocator.getPublisherEndPointAPI().findEndPointById(endpointId);
             final Optional<String> authToken = PushPublisher.retriveEndpointKeyDigest(endpoint);
-            if(!authToken.isPresent()) {
+            if (!authToken.isPresent()) {
               Logger.warn(IntegrityResource.class, "No Auth Token set for endpoint:" + endpointId);
               return response("No Auth Token set for endpoint", true);
             }
-            FormDataMultiPart form = new FormDataMultiPart();
-            form.field("AUTH_TOKEN",authToken.get());
 
-            //Sending bundle to endpoint
-            String url = endpoint.toURL()+"/api/integrity/generateintegritydata/";
+            final FormDataMultiPart form = new FormDataMultiPart();
+            form.field("AUTH_TOKEN", authToken.get());
 
-            Response response = postWithEndpointState(
-            	endpoint.getId(), url, MediaType.TEXT_PLAIN_TYPE, Entity.entity(form, form.getMediaType())
+            //Sending bundle to endpoint
+            final String url = endpoint.toURL() + "/api/integrity/generateintegritydata/";
+            final Response response = postWithEndpointState(
+                    endpoint.getId(),
+                    url,
+                    MediaType.TEXT_PLAIN_TYPE,
+                    Entity.entity(form, form.getMediaType())
             );
 
-            if(response.getStatus() == HttpStatus.SC_OK) {
+            if (response.getStatus() == HttpStatus.SC_OK) {
                 final String integrityDataRequestID = response.readEntity(String.class);
-
-                Thread integrityDataRequestChecker = new Thread() {
-
+                final Thread integrityDataRequestChecker = new Thread() {
```

Comment:
This method looks very bad I would separated in a new class as a Runnable and do some refactoring and transaction handling etc

Reply:
You are right. I wanted to move it out of the `IntegrityResource` class from the beginning but then I forgot.

Reply:
Moved.